### PR TITLE
Feature/config manager

### PR DESCRIPTION
**Pull request by mehranhadidi**
tests will be added

### CODE REVIEW

File: lib/App.js [hunk 1]
```diff
@@ -158,6 +166,11 @@ class App {
       })
     }
 
+    /** @type {ConfigManager} */
+    this._configManager = this._container.get('configManager')
```
Comment (Spomni) [inline lib/App.js]: II can't find any place where the ConfigManager would be registered. Therefore tests crash on this line.

File: lib/App.js [hunk 2]
```diff
@@ -158,6 +166,11 @@ class App {
...
+
+    console.log(this._configManager)
```
Comment (Spomni) [inline lib/App.js]: Are you sure this is necessary in the production code?

File: lib/ConfigManager.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * Merge a key with given value
+     * @param {string} key
+     * @param {Object} value
+     * @return {any}
+     */
+    merge(key, value) {
+        must.be.string(key)
+        must.be.object(value)
+        return mergeWith(value, this.get(key))
```
Comment (Spomni) [inline lib/ConfigManager.js]: Why do you use the _.mergeWith() but not _.merge() if you don't pass the customizer function to it?
Are you sure it's a good practice to merge all own and INHERITED properties of the objects?
[_.merge](https://lodash.com/docs/#merge)
[_.mergeWith](https://lodash.com/docs/#mergeWith)

**Comment by Spomni** [comment]
@shov, please answer me. It is the normal pracrice to send PR without test coverage?

**Comment by shov** [comment]
> @shov, please answer me. It is the normal practice to send PR without test coverage?

For sure that's not

**Pull request closed without merging**